cmd/util.go

```diff
@@ -84,3 +93,61 @@ func removeContents(directory string) error {
 	}
 	return nil
 }
+
+// authorized is a function decorator for authorizing RESTful
+// daemon requests. It wraps handler functions and ensures the
+// request is authorized. Returns a function
+func authorized(handler http.HandlerFunc, keyLookup func(*jwt.Token) (interface{}, error)) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		// Collect the token from the header.
+		bearerString := r.Header.Get("Authorization")
+
+		// Split out the actual token from the header.
+		splitToken := strings.Split(bearerString, "Bearer ")
+		if len(splitToken) < 2 {
+			http.Error(w, malformedAuthStringErrorMsg, http.StatusForbidden)
+			return
+		}
+		tokenString := splitToken[1]
+
+		// Parse takes the token string and a function for looking up the key.
+		token, err := jwt.Parse(tokenString, keyLookup)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusForbidden)
+			return
+		}
+
+		// Verify the claims (none for now) and token.
+		if _, ok := token.Claims.(jwt.MapClaims); !ok || !token.Valid {
+			http.Error(w, tokenInvalidErrorMsg, http.StatusForbidden)
+			return
+		}
+
+		// We're authorized, run the handler.
+		handler(w, r)
+	}
+}
+
+// generateToken creates a JSON Web Token (JWT) for a client to use when
+// sending HTTP requests to the daemon server.
+func generateToken(key []byte) (string, error) {
+	// No claims for now.
+	return jwt.New(jwt.SigningMethodHS256).SignedString(key)
+}
+
+// getAPIPrivateKey returns the private RSA key to authenticate HTTP
+// requests sent to the daemon. For now, we simply use the GitHub
+// deploy key.
+func getAPIPrivateKey(*jwt.Token) (interface{}, error) {
+	key, err := getGithubKey()
+	if err != nil {
+		return nil, err
+	}
+	return []byte(key.String()), nil
+}
+
+// getGithubKey returns the key generated by 'inertia remote bootstrap [REMOTE]'
+func getGithubKey() (ssh.AuthMethod, error) {
+	pemFile := "/app/host/.ssh/id_rsa_inertia_deploy"
```

Comment:
A bit hesitant about putting this function in `util` since that path (`/app/host/...`) applies specifically to the daemon container

Reply:
Mmmm. How about we make this function more general? `getKeyFromPem(pemFile string)`. Then we could possibly unit test it too.

Reply:
Actually I think [this](https://github.com/ubclaunchpad/inertia/blob/master/cmd/init.go#L212) is the "go" way to write testable code that does file IO. Something like `func getKeyFromPem(pemFile io.Reader)`

Reply:
I can move it back for the time being :+1: Something to think about.